### PR TITLE
Trengo Onboarding

### PR DESCRIPTION
**Pull request by utsabc**
## Trengo Onboarding

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/trengo/transform.js [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+      headers: {
+        Authorization: `Bearer ${destination.Config.apiToken}`
+      }
+    });
+  } catch (err) {
+    throw new Error("[Trengo] :: Inside lookupContact, failed to make request");
+  }
+
+  if (
+    res.status === 200 &&
```
Comment (ruchiramoitra) [inline v0/destinations/trengo/transform.js:76]: first check if res is defined or not

Comment (utsabc) [reply v0/destinations/trengo/transform.js:77]: Fixed

File: v0/destinations/trengo/transform.js [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+        "firstName"
+      )} ${getFieldValueFromMessage(message, "lastName")}`;
+
+  if (cretaeScope) {
+    // In create scope we directly create the payload for creating new contact
+    // based on the info we have.
+    let payload = {
+      name: contactName,
+      identifier,
+      channel_id: externalId || destination.Config.channelId
```
[the remark this answers is not in the document]
Comment (utsabc) [reply v0/destinations/trengo/transform.js:129]: Passing external-id through func args. If channelid is an empty string then we are checking it with the length. If it is whitespace or something wrong destination will return err response 
Example
`{
    "message": "Oops, an unexpected error occurred. Our team has been notified about this issue.",
    "errors": []
}
HTTP status 500
`

**Review by ruchiramoitra — APPROVED** [review]
LGTM

**Review by arnab-p — APPROVED** [review]

File: v0/destinations/trengo/transform.js [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+    throw new CustomError(
+      `[Trengo] :: ${event} is not present in Event-Map template keys, aborting event`,
+      400
+    );
+  }
+
+  return message.event ? hashMap[message.event] : null;
+};
+
+const stringifyJSON = json => {
```
[the remark this answers is not in the document]
Comment (utsabc) [reply v0/destinations/trengo/transform.js:53]: ok sure

File: __tests__/trengo.test.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  it(`${name} - payload: ${index}`, async () => {
+    try {
+      const output = await transformer.process(input);
+      expect(output).toEqual(expectedData[index]);
+    } catch (error) {
+      expect(error.message).toEqual(expectedData[index].error);
+    }
+  });
+});
+
```
[the remark this answers is not in the document]
Comment (utsabc) [reply __tests__/trengo.test.js:29]: yes

**Review by arnab-p — APPROVED** [review]

**Pull request merged**